src/contexts/ContextProvider.tsx

```diff
@@ -43,11 +43,13 @@ const WalletContextProvider: FC<{ endpoint?: string; children: ReactNode }> = ({
     console.error({ type: 'error', message: error.message ? `${error.name}: ${error.message}` : error.name });
   }, []);
 
+  const ShouldWrapWalletProvider = passThroughWallet ? React.Fragment : WalletProvider;
+
   return (
     <ConnectionProvider endpoint={selectedEndpoint}>
-      <WalletProvider wallets={wallets} onError={onError} autoConnect={autoConnect}>
+      <ShouldWrapWalletProvider wallets={wallets} onError={onError} autoConnect={autoConnect}>
```

Comment:
I think on top of this we need `WalletButton` to do something gracious, with this change we break the flow.

When passthroughWallet
Option 1: Pass the connect function so we can trigger showing the modal.
Option 2 (or fallback from 1): The button disappears, but then what to say to guide the user to connect wallet? "Please connect wallet from the dapp", maybe clunky

Not something that can be easily observed from the preview.

Reply:
Both option 1 & 2 is not possible,

Scenario: User not connected to dApp, and open Terminal widget
- Terminal have no way of knowing if there's existing `<WalletProvider />`
- since user's not connected, on the call

```tsx
const { wallet } = useWallet();

window.Jupiter.init({
  passThroughWallet: wallet,
})
```

the `{ wallet } = useWallet()` will still be `undefined`